interface.go

```diff
@@ -51,20 +29,25 @@ func genMethodName(ep *pb.Endpoint) string {
 
 var curlyRe = regexp.MustCompile(`^\s*{\s*(\w+)\s*<:\s*(\w+)\s*}\s*$`)
 
-func genParams(ep *pb.Endpoint) (string, error) {
+func getParams(ep *pb.Endpoint) (string, error) {
+	if ep.RestParams == nil {
+		return "", nil
+	}
 	params := make([]string, 0, 8)
 	for _, param := range ep.RestParams.QueryParam {
 		name := param.Name
-		typeStr, _, err := GenType(param.Type)
+		typeStr, _, err := GetType(param.Type)
 		if err != nil {
 			return "", err
 		}
 		matches := curlyRe.FindAllStringSubmatch(typeStr, -1)
 		if len(matches) == 1 {
 			name = matches[0][1]
 			typeStr = matches[0][2]
+			params = append(params, fmt.Sprintf("%s %s", name, typeStr))
```

Comment:
Less confusing and probably more efficient to populate two slices and splice them together at the end.